org.metaborg.meta.lang.template/syntax/layout-constraints/Layout-Constraints.sdf3

```diff
@@ -21,68 +21,73 @@ template options
-  ShortConstraint.Align = <align <ConstraintTreeReference> <{ConstraintTreeReference ", "}+>>
-  ShortConstraint.Align = <align-list <ConstraintTreeReference>>
-  ShortConstraint.NewLineIndent = <newline-indent <ConstraintTreeReference> <{ConstraintTreeReference ", "}+>>
-
-  ShortConstraint.PPOffside = <pp-offside <ConstraintTreeReference> <{ConstraintTreeReference ", "}*>>
-  ShortConstraint.PPIndent = <pp-indent <ConstraintTreeReference> <{ConstraintTreeReference ", "}+>>
-  ShortConstraint.PPAlign = <pp-align <ConstraintTreeReference> <{ConstraintTreeReference ", "}+>>
-  ShortConstraint.PPAlign = <pp-align-list <ConstraintTreeReference>>
-  ShortConstraint.PPNewLineIndent = <pp-newline-indent <ConstraintTreeReference> <{ConstraintTreeReference ", "}+>>
-  ShortConstraint.PPNewLineIndentBy = <pp-newline-indent-by(<NatCon>) <ConstraintTreeReference> <{ConstraintTreeReference ", "}+>>
-  ShortConstraint.PPNewLine = <pp-newline <ConstraintTreeReference>>
-  ShortConstraint.PPNewLineBy = <pp-newline(<NatCon>) <ConstraintTreeReference>>
-
-  ConstraintTreeReference.PosRef = NatCon
-  ConstraintTreeReference.LiteralRef = StrCon
-  ConstraintTreeReference.LabelRef = IdCon
+  Constraint.All = <all(<LayoutVar>, <Constraint>, <Constraint>)> {
+    deprecated("The 'all' layout constraint will be removed in a future version. Use align-list instead")}
+
+  Constraint.Offside = <offside <ConstraintTreeRef> <{ConstraintTreeRef ", "}*>>
+  Constraint.Indent = <indent <ConstraintTreeRef> <{ConstraintTreeRef ", "}+>>
+  Constraint.Align = <align <ConstraintTreeRef> <{ConstraintTreeRef ", "}+>>
+  Constraint.AlignList = <align-list <ConstraintTreeRef>>
+  Constraint.NewLineIndent = <newline-indent <ConstraintTreeRef> <{ConstraintTreeRef ", "}+>>
+
+  Constraint.PPOffside = <pp-offside <ConstraintTreeRef> <{ConstraintTreeRef ", "}*>>
+  Constraint.PPIndent = <pp-indent <ConstraintTreeRef> <{ConstraintTreeRef ", "}+>>
+  Constraint.PPAlign = <pp-align <ConstraintTreeRef> <{ConstraintTreeRef ", "}+>>
+  Constraint.PPAlignList = <pp-align-list <ConstraintTreeRef>>
+  Constraint.PPNewLineIndent = <pp-newline-indent <ConstraintTreeRef> <{ConstraintTreeRef ", "}+>>
+  Constraint.PPNewLineIndentBy = <pp-newline-indent-by(<NatCon>) <ConstraintTreeRef> <{ConstraintTreeRef ", "}+>>
+  Constraint.PPNewLine = <pp-newline <ConstraintTreeRef>>
+  Constraint.PPNewLineBy = <pp-newline(<NatCon>) <ConstraintTreeRef>>
 
 context-free priorities
 
-    Constraint.Not > {left:
-    Constraint.Mul
-    Constraint.Div } > {left:
-    Constraint.Add
-    Constraint.Sub} > {non-assoc:
-    Constraint.Eq
-    Constraint.Lt
-    Constraint.Le
-    Constraint.Gt
-    Constraint.Ge} > {left:
-    Constraint.And} > {left:
-    Constraint.Or},
-
-    {left: ShortConstraint.And ShortConstraint.Or}
+    Constraint.Not >
+    {non-assoc:
+      Constraint.Eq
+      Constraint.Lt
+      Constraint.Le
+      Constraint.Gt
+      Constraint.Ge
+    } >
+    {left: Constraint.And} >
```

Comment:
You don't need to annotate priority groups with a single production in the group. :)